packages/terra-dropdown-button/src/SplitButton.jsx

```diff
@@ -96,13 +137,22 @@ class SplitButton extends React.Component {
   }
 
   handleCaretKeyDown(event) {
-    if (event.keyCode === KeyCode.KEY_SPACE) {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
       this.setState({ caretIsActive: true });
+      this.setState({ openedViaKeyboard: true });
+    } else if (event.keyCode === KeyCode.KEY_DOWN && this.state.isOpen && !this.state.openedViaKeyboard) {
+      // puts focus on first list element on down arrow key press when dropdown is opened by mouse click
+      const firstOption = this.dropdownList.querySelector('[data-terra-dropdown-first-list-item = "true"]');
+      firstOption.focus();
+    } else if (event.keyCode === KeyCode.KEY_UP && this.state.isOpen && !this.state.openedViaKeyboard) {
+      // puts focus on last list element on up arrow key press when dropdown is opened by mouse click
+      const lastOption = this.dropdownList.querySelector('[data-terra-dropdown-last-list-item = "true"]');
+      lastOption.focus();
```

Comment:
What happens when there is only one item (assuming that it is a valid use-case)? Which data attribute will that item have? I'd think that this blows up.

Reply:
nice catch :)  I forgot that length will `undefined` when children itself is item and not array.  
IMO when list has only one item then it will be considered as both first and last item of the list. so updated the same by using [ React.Chlidren.count()](https://github.com/cerner/terra-core/blob/3d4cbdca1a5c96ae211cdcf5bdb5847ecfdc965c/packages/terra-dropdown-button/src/_DropdownList.jsx#L153)